sdk/monitor/monitor_client.go

```diff
@@ -43,22 +53,37 @@ func NewClient(cc *grpc.ClientConn, eventFactory EventFactory, streamConstructor
 	}
 
 	errorChannel := make(chan error, 1)
-	eventChannel := make(chan Event, 1)
+	eventChannel := make(chan Event, clientEventCapacity)
 	go func() {
 		defer close(eventChannel)
 		defer close(errorChannel)
 
-		for {
-			message, err := stream.Recv()
-			if err != nil {
-				errorChannel <- err
-				return
+		recvCh := make(chan recvResult)
```

Comment:
Would be great to have `recvCh` properly closed

Reply:
GC will free this